src/renderer/redux/actions/subscriptions.js

```diff
@@ -10,48 +10,125 @@ import type {
 import { selectSubscriptions } from 'redux/selectors/subscriptions';
 import { Lbry, buildURI } from 'lbry-redux';
 import { doPurchaseUri } from 'redux/actions/content';
-import { doNavigate } from 'redux/actions/navigation';
-import analytics from 'analytics';
+import Promise from 'bluebird';
+import Lbryio from 'lbryio';
 
 const CHECK_SUBSCRIPTIONS_INTERVAL = 60 * 60 * 1000;
 const SUBSCRIPTION_DOWNLOAD_LIMIT = 1;
 
-export const doChannelSubscribe = (subscription: Subscription) => (dispatch: Dispatch) => {
-  dispatch({
-    type: ACTIONS.CHANNEL_SUBSCRIBE,
-    data: subscription,
-  });
+const getClaimId = uri => {
+  const index = uri.indexOf('#');
+  const claimId = uri.slice(index + 1);
+  return claimId;
+};
 
-  analytics.apiLogSubscribe(subscription);
+export const doFetchMySubscriptions = () => (dispatch: Dispatch, getState: () => any) => {
+  const {
+    subscriptions: subscriptionState,
+    settings: { daemonSettings },
+  } = getState();
+  const { subscriptions: reduxSubscriptions } = subscriptionState;
+  const { share_usage_data: isSharingData } = daemonSettings;
 
-  dispatch(doCheckSubscription(subscription, true));
+  if (!isSharingData && isSharingData !== undefined) {
+    // They aren't sharing their data, subscriptions will be handled by persisted redux state
+    return;
+  }
+
+  // most of this logic comes from scenarios where the db isn't synced with redux
+  // this will happen if the user stops sharing data
+  dispatch({ type: ACTIONS.FETCH_MY_SUBSCRIPTIONS_START });
+
+  Lbryio.call('subscription', 'list')
+    .then(dbSubscriptions => {
+      const storedSubscriptions = dbSubscriptions || [];
+
+      // User has no subscriptions in db or redux
+      if (!storedSubscriptions.length && (!reduxSubscriptions || !reduxSubscriptions.length)) {
+        return [];
+      }
+
+      // There is some mismatch between redux state and db state
+      // If something is in the db, but not in redux, add it to redux
+      // If something is in redux, but not in the db, add it to the db
```

Comment:
I think this is the most reasonable choice and that we can't (or shouldn't) do anything differently immediately, but I also think it's guaranteed that this will cause a bug or confusion in the future.